include/laser_filters/intensity_filter.h

```diff
@@ -88,17 +88,21 @@ class LaserScanIntensityFilter : public filters::FilterBase<sensor_msgs::LaserSc
     {
       if (filtered_scan.intensities[i] <= lower_threshold_ ||                           // Is this reading below our lower threshold?
           filtered_scan.intensities[i] >= upper_threshold_)                             // Is this reading above our upper threshold?
-      {      
-
+      {
         filtered_scan.ranges[i] = std::numeric_limits<float>::quiet_NaN();              // If so, then make it an invalid value (NaN)
       }
 
-      if( isinf((double)filtered_scan.intensities[i]) || isnan((double)filtered_scan.intensities[i]) )
-	continue;
-      int cur_bucket = (int) ((std::min((double)filtered_scan.intensities[i], hist_max)/hist_max)*num_buckets) ;
-      if (cur_bucket >= num_buckets-1)
-	cur_bucket = num_buckets-1 ;
-      histogram[cur_bucket]++ ;
+
+      if (disp_hist_){
+        if( isinf((double)filtered_scan.intensities[i]) || isnan((double)filtered_scan.intensities[i]))
+          continue;
+        int cur_bucket = (int) std::max((std::min((double)filtered_scan.intensities[i], hist_max)/hist_max)*num_buckets, 0.0) ;
+        if( isinf(cur_bucket) || isnan(cur_bucket) )
```

Comment:
@jonbinney Sorry, I'm just confused. I updated the codes, and it's just simple rounding floating value to an int for checking if `cur_bucket` accesses out-of-index of `histogram` array or not.